keanu-project/src/test/java/io/improbable/keanu/vertices/intgr/probabilistic/MultinomialVertexTest.java

```diff
@@ -0,0 +1,302 @@
+package io.improbable.keanu.vertices.intgr.probabilistic;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import static io.improbable.keanu.tensor.TensorMatchers.allCloseTo;
+import static io.improbable.keanu.tensor.TensorMatchers.allValues;
+import static io.improbable.keanu.tensor.TensorMatchers.hasShape;
+import static io.improbable.keanu.tensor.TensorMatchers.hasValue;
+
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import io.improbable.keanu.distributions.DiscreteDistribution;
+import io.improbable.keanu.distributions.discrete.Binomial;
+import io.improbable.keanu.distributions.discrete.Categorical;
+import io.improbable.keanu.distributions.discrete.Multinomial;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.multiple.ConcatenationVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary.ReshapeVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+
+public class MultinomialVertexTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void itThrowsIfTheProbabilitiesDontSumToOne() {
+        IntegerTensor n = IntegerTensor.scalar(100);
+        DoubleTensor p = DoubleTensor.create(0., 0., 0.99, 0.).transpose();
```

Comment:
Is the transpose just to avoid having to specify a column shape?

Reply:
Yes, it's the neatest way I know to declare a column vector. Neater than this:
```
DoubleTensor.create(new double[] {0., 0., 0.99, 0.}, new int[] {4, 1})
```